app/src/main/java/com/duckduckgo/app/QueryListener.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2019 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app
+
+import androidx.appcompat.widget.SearchView
+import com.duckduckgo.app.bookmarks.db.BookmarkEntity
+import com.duckduckgo.app.bookmarks.ui.BookmarksActivity
+import com.duckduckgo.app.bookmarks.ui.BookmarksViewModel
+
+class QueryListener(
+    val adapter: BookmarksActivity.BookmarksAdapter,
+    val bookMarks: BookmarksViewModel
```

Comment:
We shouldn't have a ViewModel reference here and it may be causing a memory leak. For simplicity, let's just store the bookmarks:
`val bookmarks: List<BookmarkEntity>`

You'll need to make sure the bookmarks are kept up-to-date. If you get stuck let me know and I can provide more guidance.